### ISSUE TITLE
Alert - Ensure content updates define focus updates appropriately - (2036569779)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-components#5562]
### Module: 
 07a Alert 

### Violation: 
 Ensure content updates define focus updates appropriately 

![image](https://user-images.githubusercontent.com/5023024/202814201-c6d15feb-4fdb-4f47-8a72-6df0291a7138.png)

### WCAG Reference: 
- [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
- [2.4.3 Focus Order](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html) 

### Instance ID: 
 2036569779 

### Severity: 
 8 

### Description: 
 [Issue]
Focus does not come back to "no duration" triggering element. when actives "Close" button of alert.

[User Impact]
When content within a page changes without reloading the page and without appropriate focus changes, assistive technologies such as screen readers may not be aware of the changes or the user may only be alerted that a change in content occurred and have to review the entire page to discover exactly what changed.

[Code Reference] 
 ```
<button aria-label="Close" class="alert-close" type="button"><calcite-icon aria-hidden="true" icon="x" scale="s" calcite-hydrated=""></calcite-icon></button>
 ``` 

### Note: 
 [Suggestion]
When the context of the page changes such as when the user initiates a change that updates a large portion of the page (e.g. in a single page application) focus should move to the beginning of the updated content. Similarly, when a button is activated to open a dialog, focus should be moved into the dialog, and when the dialog is closed focus needs to return to the element that triggered the dialog.

[Compliant Code Example] 
 ```
N/A
 ```
 

### Media Type: 
 ARIA and Dynamic Content 

### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by geospatialem** [comment]
Would we be able to add context to the `calciteAlertClose` event that can restore the `:focus` to the previous element (possibly stored using the `calciteAlertBeforeOpen`)?

**Comment by geospatialem** [comment]
Closing out, the `alert` has proper focus order, and the `alert`s `setFocus` sets focus on the "close" button.
